dev-server.js

```diff
@@ -6,52 +6,108 @@ const compiler = webpack(conf)
 const express = require('express')
 const app = express()
 const bodyParser = require('body-parser')
-app.use(bodyParser.text({type: '*/*'}))
+const url = require('url')
+app.use(bodyParser.text({ type: '*/*' }))
```

Comment:
Here I change to use bodyParser.json({type: 'application/json'}), and now req.body is a proper Javascript Object.  You can now run JSON.stringify(req.body) and you will get a properly formatted JSON object, which the Perl script can successfully validate).

Reply:
Good call. Updated that locally.